engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java

```diff
@@ -33,9 +32,13 @@ public class SourcePartitionedTableTest extends RefreshingTableTestCase {
     private ExtraControlledUpdateGraph updateGraph;
     private SafeCloseable myContext;
 
-    private static final class ExtraControlledUpdateGraph extends ControlledUpdateGraph {
+    private static final class ExtraControlledUpdateGraph extends PeriodicUpdateGraph {
```

Comment:
This is still not what I was looking for; I don't want to make "real" update graphs if we can help it. The following patch is what I think we should do, although it's a lot of code:
```
Index: engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java b/engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java
--- a/engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java	(revision c1cf1bd7999ecb032b23afe4fbbe1fb86407c523)
+++ b/engine/table/src/test/java/io/deephaven/engine/table/impl/SourcePartitionedTableTest.java	(date 1694127898688)
@@ -1,27 +1,35 @@
 package io.deephaven.engine.table.impl;
 
+import io.deephaven.base.log.LogOutput;
 import io.deephaven.configuration.Configuration;
 import io.deephaven.engine.context.ExecutionContext;
 import io.deephaven.engine.table.Table;
 import io.deephaven.engine.table.impl.locations.ImmutableTableLocationKey;
 import io.deephaven.engine.table.impl.locations.TableLocationRemovedException;
 import io.deephaven.engine.table.impl.sources.ConstituentTableException;
+import io.deephaven.engine.testutil.ControlledUpdateGraph;
 import io.deephaven.engine.testutil.locations.DependentRegistrar;
 import io.deephaven.engine.testutil.locations.TableBackedTableLocationProvider;
 import io.deephaven.engine.testutil.testcase.RefreshingTableTestCase;
-import io.deephaven.engine.updategraph.impl.PeriodicUpdateGraph;
+import io.deephaven.engine.updategraph.LogicalClock;
+import io.deephaven.engine.updategraph.NotificationAdapter;
+import io.deephaven.engine.updategraph.UpdateGraph;
 import io.deephaven.engine.util.TableTools;
+import io.deephaven.io.log.LogEntry;
 import io.deephaven.io.logger.StreamLoggerImpl;
 import io.deephaven.test.types.OutOfBandTest;
 import io.deephaven.util.FindExceptionCause;
 import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.locks.AwareFunctionalLock;
 import io.deephaven.util.process.ProcessEnvironment;
 import org.jetbrains.annotations.NotNull;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static io.deephaven.engine.testutil.TstUtils.*;
 import static io.deephaven.engine.util.TableTools.col;
@@ -29,25 +37,133 @@
 @Category(OutOfBandTest.class)
 public class SourcePartitionedTableTest extends RefreshingTableTestCase {
 
-    private ExtraControlledUpdateGraph updateGraph;
-    private SafeCloseable myContext;
+    private CapturingUpdateGraph updateGraph;
+    private SafeCloseable contextCloseable;
+
+    private static final class CapturingUpdateGraph implements UpdateGraph {
 
-    private static final class ExtraControlledUpdateGraph extends PeriodicUpdateGraph {
-        final List<Runnable> sources = new ArrayList<>();
+        private final ControlledUpdateGraph delegate;
 
-        private ExtraControlledUpdateGraph() {
-            super("TEST", true, 1000, 25, -1);
+        private final ExecutionContext context;
+
+        private final List<Runnable> sources = new ArrayList<>();
+
+        private CapturingUpdateGraph(@NotNull final ControlledUpdateGraph delegate) {
+            this.delegate = delegate;
+            context = ExecutionContext.getContext().withUpdateGraph(this);
         }
 
         @Override
         public void addSource(@NotNull Runnable updateSource) {
-            super.addSource(updateSource);
+            delegate.addSource(updateSource);
             sources.add(updateSource);
         }
 
-        public void refreshSources() {
+        private void refreshSources() {
             sources.forEach(Runnable::run);
         }
+
+        @Override
+        public LogOutput append(LogOutput logOutput) {
+            return logOutput.append("CapturingUpdateGraph of ").append(delegate);
+        }
+
+        @Override
+        public boolean satisfied(final long step) {
+            return delegate.satisfied(step);
+        }
+
+        @Override
+        public UpdateGraph getUpdateGraph() {
+            return this;
+        }
+
+        private Notification wrap(@NotNull final Notification notification) {
+            return new NotificationAdapter(notification) {
+                @Override
+                public void run() {
+                    try (final SafeCloseable ignored = context.open()) {
+                        super.run();
+                    }
+                }
+            };
+        }
+
+        @Override
+        public void addNotification(@NotNull final Notification notification) {
+            delegate.addNotification(wrap(notification));
+        }
+
+        @Override
+        public void addNotifications(@NotNull final Collection<? extends Notification> notifications) {
+            delegate.addNotifications(notifications.stream().map(this::wrap).collect(Collectors.toList()));
+        }
+
+        @Override
+        public boolean maybeAddNotification(@NotNull final Notification notification, final long deliveryStep) {
+            return delegate.maybeAddNotification(wrap(notification), deliveryStep);
+        }
+
+        @Override
+        public String getName() {
+            return "CapturingUpdateGraph";
+        }
+
+        @Override
+        public AwareFunctionalLock sharedLock() {
+            return delegate.sharedLock();
+        }
+
+        @Override
+        public AwareFunctionalLock exclusiveLock() {
+            return delegate.exclusiveLock();
+        }
+
+        @Override
+        public LogicalClock clock() {
+            return delegate.clock();
+        }
+
+        @Override
+        public int parallelismFactor() {
+            return delegate.parallelismFactor();
+        }
+
+        @Override
+        public LogEntry logDependencies() {
+            return delegate.logDependencies();
+        }
+
+        @Override
+        public boolean currentThreadProcessesUpdates() {
+            return delegate.currentThreadProcessesUpdates();
+        }
+
+        @Override
+        public boolean serialTableOperationsSafe() {
+            return delegate.serialTableOperationsSafe();
+        }
+
+        @Override
+        public boolean setSerialTableOperationsSafe(final boolean newValue) {
+            return delegate.serialTableOperationsSafe();
+        }
+
+        @Override
+        public boolean supportsRefreshing() {
+            return delegate.supportsRefreshing();
+        }
+
+        @Override
+        public void requestRefresh() {
+            delegate.requestRefresh();
+        }
+
+        @Override
+        public void removeSource(@NotNull Runnable updateSource) {
+            sources.remove(updateSource);
+            delegate.removeSource(updateSource);
+        }
     }
 
     @Override
@@ -59,16 +175,13 @@
         super.setUp();
         setExpectError(false);
 
-        updateGraph = new ExtraControlledUpdateGraph();
-        updateGraph.enableUnitTestMode();
-        updateGraph.resetForUnitTests(false);
-        updateGraph.setSerialTableOperationsSafe(true);
-        myContext = ExecutionContext.getContext().withUpdateGraph(updateGraph).open();
+        updateGraph = new CapturingUpdateGraph(ExecutionContext.getContext().getUpdateGraph().cast());
+        contextCloseable = updateGraph.context.open();
     }
 
     @Override
     public void tearDown() throws Exception {
-        myContext.close();
+        contextCloseable.close();
         super.tearDown();
     }
 
@@ -122,7 +235,7 @@
         tlp.removeTableLocationKey(tlks[0]);
         tlp.refresh();
 
-        allowingError(() -> updateGraph.runWithinUnitTestCycle(() -> {
+        allowingError(() -> updateGraph.delegate.runWithinUnitTestCycle(() -> {
             updateGraph.refreshSources();
             registrar.run();
         }), errors -> errors.size() == 1 &&
@@ -135,7 +248,7 @@
 
         tlp.addPending(p3);
         tlp.refresh();
-        updateGraph.runWithinUnitTestCycle(() -> {
+        updateGraph.delegate.runWithinUnitTestCycle(() -> {
             updateGraph.refreshSources();
             registrar.run();
         });
@@ -148,7 +261,7 @@
         tlp.removeTableLocationKey(tlks[0]);
         tlp.refresh();
 
-        allowingError(() -> updateGraph.runWithinUnitTestCycle(() -> {
+        allowingError(() -> updateGraph.delegate.runWithinUnitTestCycle(() -> {
             updateGraph.refreshSources();
             registrar.run();
         }), errors -> errors.size() == 1 &&
@@ -163,8 +276,8 @@
         // The TableBackedTableLocation has a copy() of the p3 table which is itself a leaf. Erroring P3 will
         // cause one error to come from the copied table, and one from the merged() table. We just need to validate
         // that the exceptions we see are a ConstituentTableException and an ISE
-        allowingError(() -> updateGraph.runWithinUnitTestCycle(
-                () -> p3.notifyListenersOnError(new IllegalStateException("This is a test error"), null)),
+        allowingError(() -> updateGraph.delegate.runWithinUnitTestCycle(
+                        () -> p3.notifyListenersOnError(new IllegalStateException("This is a test error"), null)),
                 errors -> errors.stream()
                         .anyMatch(e -> FindExceptionCause.findCause(e,
                                 IllegalStateException.class) instanceof IllegalStateException)
```